### PR TITLE
 storage_backend_sftp: refactor tests w/ mock

### PR DESCRIPTION
**Pull request by simahawk**
Includes fix for #7
ping @lmignon @gurneyalex

### CODE REVIEW

**Comment by simahawk** [comment]
weird, pre-commit fails but ran it locally.
Ah, maybe better to wait for https://github.com/OCA/storage/pull/12

**Review by lmignon — APPROVED** [review]
Thank you @simahawk LGTM

**Comment by lmignon** [comment]
> weird, pre-commit fails but ran it locally.
> Ah, maybe better to wait for #12

unfortunately I don't have the right to merge PRs into this repos...

File: storage_backend_sftp/components/sftp_adapter.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def get(self, relative_path, **kwargs):
         with sftp(self.collection) as client:
             file_data = client.open(full_path, "rb")
             data = file_data.read()
+            # TODO: shouldn't we close the file?
```
Comment (simahawk) [inline storage_backend_sftp/components/sftp_adapter.py:87]: @lmignon what do you think about this?

**Comment by simahawk** [comment]
Finally green, runbot is happy now :)

@lmignon we can merge this and discuss later about the TODO https://github.com/OCA/storage/pull/16#discussion_r336468903

**Comment by lmignon** [comment]
IMO yes we can. Unfortunately, I don't have the merge rights yet.

ping @sebastienbeau

**Review by lmignon — APPROVED** [review]

**Comment by simahawk** [comment]
@lmignon maybe today you have the rights? :)
Otherwise we can ping @sbidoul :stuck_out_tongue:

**Pull request merged**